packages/x-components/src/x-modules/scroll/components/__tests__/scroll-to-top.spec.ts

```diff
@@ -1,60 +1,63 @@
 import { mount, Wrapper } from '@vue/test-utils';
 import Vue from 'vue';
-import { installNewXPlugin } from '../../../__tests__/utils';
-import { XEvent, XEventPayload } from '../../../wiring/events.types';
-import BaseScrollToTop from '../base-scroll-to-top.vue';
+import { installNewXPlugin } from '../../../../__tests__/utils';
+import { XEvent, XEventPayload } from '../../../../wiring/events.types';
+import { scrollXModule } from '../../x-module';
+import ScrollToTop from '../scroll-to-top.vue';
 
 /**
- * Renders the {@link BaseScrollToTop} with the provided options.
+ * Renders the {@link ScrollToTop} with the provided options.
  *
  * @param options - The options to render the component with.
  * @returns An small API to test the component.
  */
-function renderBaseScrollToTop({
+function renderScrollToTop({
   defaultSlot = '<span>Top</span>',
   scrollId = 'scrollId',
   thresholdPx
-}: RenderBaseScrollToTopOptions = {}): RenderBaseScrollToTopAPI {
-  const [, localVue] = installNewXPlugin();
-  const wrapper = mount(BaseScrollToTop, {
+}: RenderScrollToTopOptions = {}): RenderScrollToTopAPI {
+  const [, localVue] = installNewXPlugin({ initialXModules: [scrollXModule] });
+  const wrapper = mount(ScrollToTop, {
     propsData: { scrollId, thresholdPx },
     localVue,
     scopedSlots: {
       default: defaultSlot
     }
   });
 
-  const scrollToTopWrapper = wrapper.findComponent(BaseScrollToTop);
+  const scrollToTopWrapper = wrapper.findComponent(ScrollToTop);
 
   return {
     scrollToTopWrapper,
-    async click() {
+    click() {
       scrollToTopWrapper.trigger('click');
-      await localVue.nextTick();
+      return localVue.nextTick();
     },
-    async emitXEvent<Event extends XEvent>(event: Event, payload: XEventPayload<Event>) {
+    emitXEvent(event, payload) {
       scrollToTopWrapper.vm.$x.emit(event, payload, { id: 'scrollId' });
-      await scrollToTopWrapper.vm.$nextTick();
+      return localVue.nextTick();
     }
   };
 }
 
-describe('testing Base Scroll To Top component', () => {
+describe('testing Scroll To Top component', () => {
   it('renders the content in the slot', async () => {
-    const { scrollToTopWrapper, emitXEvent } = renderBaseScrollToTop();
-    await emitXEvent('UserAlmostReachedScrollEnd', 100);
+    const { scrollToTopWrapper, emitXEvent } = renderScrollToTop();
+    await emitXEvent('UserAlmostReachedScrollEnd', true);
+    await emitXEvent('UserChangedScrollDirection', 'DOWN');
     expect(scrollToTopWrapper.text()).toEqual('Top');
   });
 
   it('shows if a scroll is almost reaching the end and there is no threshold', async () => {
-    const { scrollToTopWrapper, emitXEvent } = renderBaseScrollToTop();
+    const { scrollToTopWrapper, emitXEvent } = renderScrollToTop();
     expect(scrollToTopWrapper.html()).toBe('');
```

Comment:
Is confusing, at least for me, that when the component is not visible it's rendered.

I mean, I would expect to test something like `expect(scrollToTopWrapper).toBeUndefined();` instead of an empty string.

Reply:
I don't think this confusing. Maybe a `expect(xxx).toRenderNothing()` would be more readable, but reading that a component is rendering no HTML is fine for me. In any case, that's Vue test utils problem, which forces you to test it like this. And you can't use `expect(scrollToTopWrapper.element).toBeUndefined()` because the element is actually a comment node, so it is not undefined.